internal/util/prometheus.go

```diff
@@ -0,0 +1,55 @@
+package util
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+)
+
+type ControllerFunctionalPrometheusMetrics struct {
+	// number of post /config to proxy successfully
+	ConfigPassCounter prometheus.Counter
+	// number of post /config to proxy failed
+	ConfigFailureCounter prometheus.Counter
+	// number of ingress analysis failure
+	ParseFailureCounter prometheus.Counter
```

Comment:
Consider naming this metric `translate_count` - where each parsing event would increment the counter - with the following label:
- `success` with possible values `true`, `false`

Reply:
I think you mean to comment following line 
https://github.com/Kong/kubernetes-ingress-controller/pull/1622/files#diff-daa0c4228afcca49d1ea0f7b23f509728c90ce7d8f9748b37dbfd6ae6b2598b4R44

right ?

Reply:
According to Prometheus official documentation  https://prometheus.io/docs/concepts/data_model/#metric-names-and-labels
It is recommended "The metric name specifies the general feature of a system that is measured",  and this is exactly to measure our controller ingress parse sub-system.